pvlib/iotools/bsrn.py

```diff
@@ -4,56 +4,367 @@
 
 import pandas as pd
 import gzip
+import ftplib
+import warnings
+import io
+import os
 
-COL_SPECS = [(0, 3), (4, 9), (10, 16), (16, 22), (22, 27), (27, 32), (32, 39),
-             (39, 45), (45, 50), (50, 55), (55, 64), (64, 70), (70, 75)]
+BSRN_FTP_URL = "ftp.bsrn.awi.de"
 
-BSRN_COLUMNS = ['day', 'minute',
-                'ghi', 'ghi_std', 'ghi_min', 'ghi_max',
-                'dni', 'dni_std', 'dni_min', 'dni_max',
-                'empty', 'empty', 'empty', 'empty', 'empty',
-                'dhi', 'dhi_std', 'dhi_min', 'dhi_max',
-                'lwd', 'lwd_std', 'lwd_min', 'lwd_max',
-                'temp_air', 'relative_humidity', 'pressure']
+BSRN_LR0100_COL_SPECS = [(0, 3), (4, 9), (10, 16), (16, 22), (22, 27),
+                         (27, 32), (32, 39), (39, 45), (45, 50), (50, 55),
+                         (55, 64), (64, 70), (70, 75)]
 
+BSRN_LR0300_COL_SPECS = [(0, 3), (4, 9), (10, 16), (16, 22), (22, 27)]
 
-def read_bsrn(filename):
+BSRN_LR0500_COL_SPECS = [(0, 3), (3, 8), (8, 14), (14, 20), (20, 26), (26, 32),
+                         (32, 38), (38, 44), (44, 50), (50, 56), (56, 62),
+                         (62, 68), (68, 74), (74, 80)]
+
+BSRN_LR0100_COLUMNS = ['day', 'minute',
+                       'ghi', 'ghi_std', 'ghi_min', 'ghi_max',
+                       'dni', 'dni_std', 'dni_min', 'dni_max',
+                       'empty', 'empty', 'empty', 'empty', 'empty',
+                       'dhi', 'dhi_std', 'dhi_min', 'dhi_max',
+                       'lwd', 'lwd_std', 'lwd_min', 'lwd_max',
+                       'temp_air', 'relative_humidity', 'pressure']
+
+BSRN_LR0300_COLUMNS = ['day', 'minute', 'upward short-wave reflected',
+                       'upward long-wave', 'net radiation']
+
+BSRN_LR0500_COLUMNS = ['day', 'minute', 'uva_global_mean', 'uva_global_std',
+                       'uva_global_min', 'uva_global_max', 'uvb_direct_mean',
+                       'uvb_direct_std', 'uvb_direct_min', 'uvb_direct_max',
+                       'empty', 'empty', 'empty', 'empty',
+                       'uvb_global_mean', 'uvb_global_std', 'uvb_global_min',
+                       'uvb_global_max', 'uvb_diffuse_mean', 'uvb_diffuse_std',
+                       'uvb_diffuse_mean', 'uvb_diffuse_std',
+                       'uvb_diffuse_min', 'uvb_diffuse_max',
+                       'uvb_reflect_mean', 'uvb_reflect_std',
+                       'uvb_reflect_min', 'uvb_reflect_max']
+
+
+def get_bsrn(start, end, station, username, password, logical_records=['0100'],

```

Comment:
I actually have it right now so it can either take a list or string. What do you think of that?
`    logical_records: str or list, default: ['0100']`

Therefore, the default could just be the string version: '0100'.

Otherwise, should I also change the input type in the doc string to tuple?